tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec

```diff
@@ -29,19 +32,54 @@
                 <dependency id="Newtonsoft.Json" version="13.0.1"/>
                 <dependency id="System.Collections.Immutable" version="1.7.1"/>
             </group>
+            <!--TODO - why do we reference nunit from dynamo core? This is an odd partial collection of dependencies to mark.-->
```

Comment:
Any answer to this question?

Reply:
This is the full dependency list generated by dotnet
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>DynamoCore</id>
    <version>1.0.0</version>
    <authors>DynamoCore</authors>
    <description>Package Description</description>
    <dependencies>
      <group targetFramework=".NETFramework4.8">
        <dependency id="DynamoUtilities" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="GraphLayout" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoAssociative" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoCore" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoImperative" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoScript" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="DesignScriptBuiltin" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="DynamoUnits" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="VMDataBridge" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="DynamoServices" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="Autodesk.IDSDK" version="1.1.6" exclude="Build,Analyzers" />
        <dependency id="DynamoVisualProgramming.Analytics" version="3.0.1.740" exclude="Build,Analyzers" />
        <dependency id="DynamoVisualProgramming.LibG_228_0_0" version="2.18.0.1400" exclude="Build,Analyzers" />
        <dependency id="DynamoVisualProgramming.LibG_229_0_0" version="2.18.0.1400" exclude="Build,Analyzers" />
        <dependency id="Greg" version="2.5.0.5076" exclude="Build,Analyzers" />
        <dependency id="Lucene.Net" version="4.8.0-beta00016" exclude="Build,Analyzers" />
        <dependency id="Lucene.Net.Analysis.Common" version="4.8.0-beta00016" exclude="Build,Analyzers" />
        <dependency id="Lucene.Net.QueryParser" version="4.8.0-beta00016" exclude="Build,Analyzers" />
        <dependency id="Newtonsoft.Json" version="13.0.1" exclude="Build,Analyzers" />
        <dependency id="RestSharp" version="106.12.0" exclude="Build,Analyzers" />
      </group>
    </dependencies>
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="Microsoft.CSharp" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="Microsoft.Win32.Primitives" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.AppContext" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Configuration" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Console" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Diagnostics.Tracing" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Globalization.Calendars" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.Compression" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.Compression.ZipFile" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.FileSystem" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.FileSystem.Primitives" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Linq" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Linq.Expressions" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Net.Http" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Net.Sockets" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Reflection" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime.Extensions" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime.InteropServices" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime.InteropServices.RuntimeInformation" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.Algorithms" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.Encoding" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.Primitives" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.X509Certificates" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Text.RegularExpressions" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Xml.ReaderWriter" targetFramework=".NETFramework4.8" />
    </frameworkAssemblies>
  </metadata>
</package>
```
I am not sure anyone will find the partial list useful.

Reply:
1. hmm thanks @pinzart90 - can you tell me how you generated that list?

2. If someone will find the partial list useful or not: that ultimately depends on what the use case is, and for simply writing nodeModel nodes - users really only need `system.collections.immutable` and `newtonsoft` references present to compile correctly.

I am not sure why users would need Nunit present for DynamoCore consumption.

Again - if someone was actually going to use this package to embed DynamoModel into something then they would definitely be helped by having your full list.

BUT having that full list will make the first use case of writing nodes for Dynamo very difficult.

3. so here is what I am thinking:

* I do not wish to include this full list until these packages are transitioned to `ref` folder.
* In the next PR I will look at moving all current packages to ref, and including full dep lists so that all reference assemblies that could be used in public APIs are available at compile time.
* I will file a task to consider the embedding of `DynamoModel` as a separate use case from `DynamoVisualProgrmaming.Core` and to create a separate package with full dep list and using the `lib` folder. We can prioritize that later.